[1LP][RFR] Vm name enhancement - to append assignee to VM name

kedark3 opened the pull request:
## Purpose or Intent

- __Enhancement__ for feature vm.create_on_provider() method to take request fixture, fetch assignee from polarion docblock and postfix assignee to VM name

**Tested all affected files on jenkins 511z/job/kk-console-chrome/69/ where I replaced all full_template_vm usage with new create_vm_fixture and it passed correctly. Please review.**

### PRT Run


- {{ pytest: cfme/utils/tests/test_vm_name_assignee.py cfme/utils/tests/test_vm_name_assignee_func_scope.py  --use-provider vsphere67-nested  -v }}

kedark3 commented:
@jawatts @mshriver this is my initial thought around how to get VM names postfixed with asignee. Tested it locally and it worked fine. Let me know how can we improve it. Thanks.

dajoRH commented:
I detected some fixture changes in commit 245f36409046329d80442f2ebb3dac99c219ef21

The *local* fixture **`vm_obj`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_html5_vm_console.py**
    - *test_html5_vm_console*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

john-dupuy commented on cfme/common/vm.py:
```diff
@@ -792,7 +793,16 @@ def create_on_provider(self, timeout=900, find_in_cfme=False, delete_on_failure=
-        vm = deploy_template(self.provider.key, self.name, self.template_name, **kwargs)
+        try:
+            import re
+            assignee = re.findall(r'assignee: *\w+',
```
Would it be possible to use the `polarion-docstrings` library to get this data? We could potentially use `polarion_data = parser(request.function)`, and then just have `assignee = polarion_data.assignee`. https://gitlab.com/mkourim/polarion_docstrings/blob/master/polarion_docstrings/parser.py

john-dupuy replied:
Here's a small example, after importing `DocstringParser` from `polarion_docstring.parser` 
```python,
In[17]: DocstringParser(test_alert_hardware_reconfigured.__doc__).parse()["assignee"].value
Out[17]: 'jdupuy'
```

kedark3 replied:
@john-dupuy I updated this.

john-dupuy commented on cfme/common/vm.py:
```diff
@@ -792,7 +793,15 @@ def create_on_provider(self, timeout=900, find_in_cfme=False, delete_on_failure=
         """
-        vm = deploy_template(self.provider.key, self.name, self.template_name, **kwargs)
+        try:
+            from polarion_docstrings.parser import DocstringParser
```
I would move this import to the top of the file.

john-dupuy commented on cfme/common/vm.py:
```diff
@@ -792,7 +793,15 @@ def create_on_provider(self, timeout=900, find_in_cfme=False, delete_on_failure=
-        vm = deploy_template(self.provider.key, self.name, self.template_name, **kwargs)
+        try:
+            from polarion_docstrings.parser import DocstringParser
+            assignee = DocstringParser(request.function.__doc__).parse()["assignee"].value
```
`request` defaults to `None`, and I have seen many usages of `create_on_provider` that do not pass request. I think this should be checked.

john-dupuy replied:
Oh I see, you're catching the `TypeError` from request being `None`

john-dupuy replied:
I think this `try/except` block should be changed to a `if request:` condition. What do you think?

john-dupuy commented on cfme/common/vm.py:
```diff
@@ -792,7 +793,15 @@ def create_on_provider(self, timeout=900, find_in_cfme=False, delete_on_failure=
-        vm = deploy_template(self.provider.key, self.name, self.template_name, **kwargs)
+        try:
+            from polarion_docstrings.parser import DocstringParser
+            assignee = DocstringParser(request.function.__doc__).parse()["assignee"].value
```
Perhaps change to a `.get()` call for the `assignee`?

jawatts replied:
with a default of `""`

john-dupuy commented on cfme/tests/cloud_infra_common/test_html5_vm_console.py:
```diff
@@ -44,7 +44,7 @@ def vm_obj(request, provider, setup_provider, console_template):
     request.addfinalizer(lambda: vm_obj.cleanup_on_provider())
 
-    vm_obj.create_on_provider(timeout=2400, find_in_cfme=True, allow_skip="default")
+    vm_obj.create_on_provider(request, timeout=2400, find_in_cfme=True, allow_skip="default")
```
Do you want to also search for other usages of `create_on_provider` and add `request` where it can be added? What about fixtures and functions that do not currently depend on request?

jawatts commented on cfme/common/vm.py:
```diff
@@ -792,7 +793,15 @@ def create_on_provider(self, timeout=900, find_in_cfme=False, delete_on_failure=
+        try:
+            from polarion_docstrings.parser import DocstringParser
+            assignee = DocstringParser(request.function.__doc__).parse()["assignee"].value
+        except TypeError:
```
we can remove this is we use the `.get()`

kedark3 commented:
@john-dupuy I had chat with @mshriver he had some thoughts around moving it to Fixtures layer, he will comment on the PR shortly and I will re-work it then :) Thanks for review.

mshriver commented on cfme/common/vm.py:
```diff
@@ -792,7 +793,15 @@ def create_on_provider(self, timeout=900, find_in_cfme=False, delete_on_failure=
-        vm = deploy_template(self.provider.key, self.name, self.template_name, **kwargs)
+        try:
+            from polarion_docstrings.parser import DocstringParser
+            assignee = DocstringParser(request.function.__doc__).parse()["assignee"].value
```
This metadata can be defined at levels other than the function. There are new methods which merge the meta from all levels, in a given file. I'm using them in #9225

mshriver commented on cfme/fixtures/vm.py:
```diff
@@ -11,7 +12,15 @@
     collection = provider.appliance.provider_based_collection(provider)
-    vm_obj = collection.instantiate(vm_name, provider, template_name=template.name)
+    try:
+        assignee = DocstringParser(request.function.__doc__).parse()["assignee"].value
```
With the most recent versions of the polarion tools (PR #9225 includes update) this metadata (composited from module/class/function) is available from the session.

Currently that session attribute is only populated if in collect only mode, I'm looking to change that in pytest-polarion-collect PR 5 or a followup.

This works for now for testing this functionality and designing the fixture structure, so I don't think it blocks working the rest of this PR.  But ASAP this should be changed to use the pre-parsed (and composited) metadata.

kedark3 replied:
I have access to items and fspath through the session object. I do see nodeid too but that returns blank. I don't see what we can use to fetch the metadata as you mentioned in the comment above. I agree that once it becomes available we can use it. 
```
n [15]: s.items                                                                                                                                                                                                                               
Out[15]: [<Function 'test_vm_name_postfix[virtualcenter-6.7-full_template]'>]

In [16]: s.fspath                                                                                                                                                                                                                              
Out[16]: local('/home/kkulkarn/git/integration_tests')

In [17]: s.nodeid                                                                                                                                                                                                                              
Out[17]: ''
```

mshriver commented on cfme/fixtures/vm.py:
```diff
@@ -11,7 +12,15 @@
+    except TypeError:
+        assignee = ""
+    if assignee:
+        vm_name = vm_name + "-" + assignee
```
I think the logic can be simplified a bit here to avoid adding these strings in a clunky way, and do it more consistently.

Consider using `'unassigned'` or something similar when its missing.

mshriver commented on cfme/fixtures/vm.py:
```diff
@@ -58,13 +67,29 @@ def _get_vm_name(request):
+        template_type = request.param
     vm_name = _get_vm_name(request)
-    return _create_vm(request, full_template, provider, vm_name)
+    return _create_vm(request, template_type, provider, vm_name)
```
Obviously need some defaulting behavior here, whether its test uncollect/skip, or trying to pick `'small_template'` from the yaml.

kedark3 replied:
skipping the test here makes sense, IMHO, cause if you are running a test with some template, you definitely need that template, defaulting to another one will not work. This is specially true for vm console tests I know, and may be applicable to others too.

dajoRH commented:
I detected some fixture changes in commit 0ab1b5233dea8b51e9c5046adc1257e1eb7d41bc

The *global* fixture **`create_vm_modscope`** is used in the following files:

The *global* fixture **`create_vm`** is used in the following files:

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit f5c9ef8aa52fd9d33814497fa4bef59b61bb0cc6

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/cloud/test_providers.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit b1a057a732b22f76be2ba9f992eae43008c3c89f

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/cloud/test_providers.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit 7a48cace23404c140bbb94e09686244e426e3a4a

The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_vm_name_postfix.py**
    - *test_vm_name_postfix_1*
    - *test_vm_name_postfix_2*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/cloud/test_providers.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit 5bc285ac07e8fc9bf588236db4ec72e51e29dfdc

The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_vm_name_postfix.py**
    - *test_vm_name_postfix_1*
    - *test_vm_name_postfix_2*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/cloud/test_providers.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit 31663b38a3db7e53feb576ad9f6179b9267be501

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/cloud/test_providers.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

kedark3 commented:
Run Number 34335 passed successfully. After that I removed test cases `test_vm_name_postfix.py` after that to make PR ready for merge. Hence following run failed.

jawatts commented on cfme/fixtures/vm.py:
```diff
@@ -10,12 +14,49 @@
+                assignee_list.append(merged_str['::'.join([request.node.fspath.strpath, test])]
+                    .get('assignee', ''))
+            except KeyError:
+                continue
```
does this need to be in a `Try/Except` block? If you are doing a .`get()` it should not throw a `KeyError`?

kedark3 replied:
`merged_str['::'.join([request.node.fspath.strpath, test]` this could throw `KeyError` right?

digitronik commented on cfme/fixtures/vm.py:
```diff
@@ -10,12 +14,49 @@
+        for test in test_list:
+            try:
+                assignee_list.append(merged_str['::'.join([request.node.fspath.strpath, test])]
+                    .get('assignee', ''))
```
optional:
It will be good to put default `None`

digitronik commented on cfme/fixtures/vm.py:
```diff
@@ -10,12 +14,49 @@
+        # If all tests have same assignee, set length will be 1, else set assignee='module'
+        assignee = assignee_list[0] if len(set(assignee_list)) == 1 else 'module'
+
+    vm_name = "-".join([vm_name, assignee])
```
lets use `f-string` like
```
vm_name = "{vm_name}-{assignee}"
```

dajoRH commented:
I detected some fixture changes in commit 4bda776d406b038d6f38b5a54095d0ab52f084ed

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/cloud/test_providers.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

mshriver commented on cfme/fixtures/vm.py:
```diff
@@ -10,12 +14,49 @@
 
 
 def _create_vm(request, template, provider, vm_name):
+    doc_str = parser.get_docstrings_in_file(request.node.fspath.strpath)
```
I would look at pytest-polarion-collect's `get_parsed_docstring`, which takes (and populates) a docstring cache.

The issue here is that we're going to parse docstrings repeatedly for every function/module scoped fixture using this method.

You can access the session through request, and create a docstrings attribute there to act as a cache of the parsed docstrings as files are read.

I would like to setup pytest-polarion-collect to generate this cache always, not just when one of the kwargs is being used (collect-only, generate-json, or meta-filter).  It's a startup cost, but then any fixtures trying to use the meta will get it for free.

kedark3 commented:
https://github.com/ManageIQ/integration_tests/pull/9186#issuecomment-528559486 -- that comment is incorrect. File mentioned in there does not use the fixture.

dajoRH commented:
I detected some fixture changes in commit 4bda776d406b038d6f38b5a54095d0ab52f084ed

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/cloud/test_providers.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
Lint report for commit 25cff89cdc3e2d70d2df60220a995bfb34e38e39:

`cfme/utils/tests/test_vm_name_assignee.py`:
- :heavy_exclamation_mark: Line 71:59: **W292** *no newline at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

dajoRH commented:
Lint report for commit 5b88dcfa1ab9e8f858ce90fcf925de5ecaf37257:

`cfme/utils/tests/test_vm_name_assignee.py`:
- :heavy_exclamation_mark: Line 71:59: **W292** *no newline at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

dajoRH commented:
I detected some fixture changes in commit 5b88dcfa1ab9e8f858ce90fcf925de5ecaf37257

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/cloud/test_providers.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit 701a43fc04e469f4fe853c2e511e304fa462b422

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/cloud/test_providers.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
Lint report for commit 701a43fc04e469f4fe853c2e511e304fa462b422:

`cfme/utils/tests/test_vm_name_assignee.py`:
- :heavy_exclamation_mark: Line 71:59: **W292** *no newline at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

dajoRH commented:
I detected some fixture changes in commit b7b8349fdd28c80d23efe909bc2e7524514d9f10

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

dajoRH commented:
I detected some fixture changes in commit 930d865ac4e143ed6420f92237e5d8c78cebb113

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit fdd043a7fbfaf7cdbf65918490afecbf3f4d7747

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit 5b6fc5c46a34271eff0fb1c908984800cde9a3c2

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
Lint report for commit 5b6fc5c46a34271eff0fb1c908984800cde9a3c2:

`cfme/utils/tests/test_vm_name_assignee.py`:
- :bangbang: Line 12:101: **E501** *line too long (101 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

dajoRH commented:
I detected some fixture changes in commit f8ac51823c377925245379191c1e2f2a3450b4c7

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit 0410e65b07d3cfa31074d64f295ae9828aef410f

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

jawatts commented on cfme/fixtures/vm.py:
```diff
@@ -9,13 +13,68 @@
+
+def pytest_addoption(parser):
+    parser.addoption(
+        '--no-assignee-vm-name',
```
maybe something shorty here like `--no-vm-name`? See what other people think.

kedark3 replied:
I think we are keeping this as is for now unless @mshriver wants to change it to `--no-vm-auto-naming` ??

mshriver replied:
We'll just keep it and collectively hate it forever.

dajoRH commented:
I detected some fixture changes in commit 00e6140abad6545eafd21ac4ac8f5282fa11e45d

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

jawatts commented on cfme/fixtures/vm.py:
```diff
@@ -9,13 +13,68 @@
 
 
+DOCSTRING_CACHE = {}
+TEST_PARAMS = re.compile(r"\[.*\]")
```
any reason this is defined up here but only used once?

jawatts commented on cfme/fixtures/vm.py:
```diff
@@ -9,13 +13,68 @@
+            test_list = [
+                item
+                for item in dir(request.module)
+                if ('test_' in item[:5]) and not ('test_requirements' == item)
```
Can you add a comment why you are taking items 0-5?

mshriver replied:
Looks like this is just an odd way of expressing `item.startswith('test_')`

dajoRH commented:
I detected some fixture changes in commit f51f8912e4e0931c2b2e2ca970508741ffe5d0a6

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit e17352de8917d3fb78cf292563c3fd7c23034760

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit d2214cfa23b0045703067a54389e427ab2e1a0d4

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit 8d69c16a1c944a6d3834398ec20997afaf853dcf

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

mshriver commented on cfme/fixtures/vm.py:
```diff
@@ -1,44 +1,74 @@
+
+@pytest.hookimpl(trylast=True)
+def pytest_collection_modifyitems(session, config, items):
+    if not config.getoption('--no-assignee-vm-name'):
```
Please include a docblock for this.

Describe that this proces_json_data is loading a cache in the pytest session with the metadata.

mshriver commented on requirements/frozen.txt:
```diff
@@ -257,7 +257,7 @@ pyparsing==2.4.0
 pytesseract==0.3.0
 pytest==5.2.1
-pytest-polarion-collect==0.21.0
+pytest-polarion-collect==0.22.0
```
Please re-freeze, Martin just released 0.23

mshriver commented on cfme/utils/tests/test_vm_name_assignee_func_scope.py:
```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
```
Thanks for writing some unit tests for this! Much appreciated.

mshriver commented on cfme/fixtures/vm.py:
```diff
@@ -1,44 +1,74 @@
+
+
+@pytest.hookimpl(trylast=True)
+def pytest_collection_modifyitems(session, config, items):
```
Discussed offline - there's no collection modification happening here, it would be better to call process_json_data in `pytest_collection_finish`.

dajoRH commented:
I detected some fixture changes in commit 7f8149de2c42738aad846bdc76e1adc00b208d73

 The *global* fixture **`create_vm_modscope`** was changed, but I didn't find where it's used.
 The *global* fixture **`create_vm`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

mshriver commented on cfme/fixtures/vm.py:
```diff
@@ -1,44 +1,76 @@
         vm_obj.cleanup_on_provider()
         provider.refresh_provider_relationships()
 
+    vm_obj.mgmt.ensure_state(VmState.RUNNING)
```
Why remove the net_check and ip_address wait_for calls in the fixture currently?

kedark3 replied:
As discussed offline, not all templates have ability to share the Pingable IPs as sometimes some of them do not have Guest tools installed and not all the tests need these either, so having a check to test IP etc can cause failure even if test case calling this fixture doesn't care for IP.

mshriver commented on cfme/fixtures/vm.py:
```diff
@@ -59,12 +91,20 @@ def _get_vm_name(request):
 
 
 @pytest.fixture(scope="module")
-def full_template_vm_modscope(setup_provider_modscope, request, full_template_modscope, provider):
```
These fixtures are currently in use in the framework, and can't just be removed.

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

dajoRH commented:
Lint report for commit 6a61cd7e691d486397501801347cde173d19598a:

`cfme/fixtures/vm.py`:
- :red_circle: Line 108:9: **F841** *local variable 'template_type' is assigned to but never used*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

dajoRH commented:
I detected some fixture changes in commit 6a61cd7e691d486397501801347cde173d19598a

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit bf198015e832637c3ebaa5d0c8f98380b32c80e2

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

kedark3 commented:
Tested this PR on jenkins `511z/job/kk-console-chrome/69/` where I replaced all `full_template_vm` usage with new `create_vm_fixture` and it passed correctly. Please review.

dajoRH commented:
I detected some fixture changes in commit 0d4bcbf3cb571534b08857fa330eed0daa7255eb

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

dajoRH commented:
I detected some fixture changes in commit a63e4cb0a691f0129ecff5ca47a2c1cf4ee47bfa

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

mshriver commented:
@kedark3 the changes looked good, but its failing with dummy appliance running in travis - need to address that situation of why we hit a KeyError in pytest-polarion-collect

dajoRH commented:
I detected some fixture changes in commit 8cdf73644321d076e8bf466689642d453ee9552c

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

kedark3 commented:
@mshriver the KeyError is occurring in following places: 
```
cfme/modeling/tests/test_collections.py
cfme/scripting/tests/test_quickstart.py
```
(I stopped debugging it after that, just removed `all_items=True`, please read below)

And that is due to lack of polarion_docblock in those files. That is caused by the argument `all_items=True` that I added after martin made the enhancement, so that I can allow parsing and testing this vm_name enhancement. 
```
cfme/utils/tests/test_vm_name_assignee.py 
cfme/utils/tests/test_vm_name_assignee_func_scope.py 
```


If I remove the `all_items=True`, it will no longer attempt to parse files in directories that it shouldn't(i.e. where `.polarion_tests` file is missing) , causing above two tests in `utils/tests/` to not work as expected. We can move those tests elsewhere if we indeed want to make them work.

After I removed the `all_items=True` and added dummy Polarion docblock in the `cfme/tests/networks/nuage/test_automation.py` then I was able to run the collection locally and also in Travis.
Please see last commit on the PR.

mshriver commented:
@kedark3 noted about the interaction between the test functions, the all_items option, and the polarion docblock.

Simply adding polarion docblocks to the network tests missing them leaves a hole though - There is no longer a flake8 item for missing polarion docblock (we were ignoring it before anyway). So a test added, with no polarion docblock, would cause test collection failures again.

mshriver commented:
@kedark3 running locally, it looks like there is no need for the empty polarion docblocks.

dajoRH commented:
I detected some fixture changes in commit cd3a4787e7c6fe5091eda6f9374370c2b4072539

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit ab664fab4ebaa42566bea7590e467cdcaa9be0bc

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

dajoRH commented:
Lint report for commit be1ccbb116f547f9b00d613e9d309182e1e58097:

`cfme/utils/tests/test_vm_name_assignee.py`:
- :heavy_exclamation_mark: Line 25:1: **W293** *blank line contains whitespace*
- :heavy_exclamation_mark: Line 45:1: **W293** *blank line contains whitespace*

`cfme/utils/tests/test_vm_name_assignee_func_scope.py`:
- :heavy_exclamation_mark: Line 24:1: **W293** *blank line contains whitespace*
- :heavy_exclamation_mark: Line 44:1: **W293** *blank line contains whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

dajoRH commented:
I detected some fixture changes in commit be1ccbb116f547f9b00d613e9d309182e1e58097

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit ab664fab4ebaa42566bea7590e467cdcaa9be0bc

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit a9e368c80074011c79588b348f4019a04db50824

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit 494c33894749049db0f47fe596f74d2ab77e01f9

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit a3d7a8d26f5b0f277f70f51882b3e93fc5f2a5a3

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

dajoRH commented:
I detected some fixture changes in commit f2c6b733ee2b703cac165ab941ad34ed425d6e34

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

kedark3 commented:
@jawatts @mshriver Travis failures are now resolved. PR is passing locally when ran against Wharf. PRT should pass too. :crossed_fingers: Kindly review.

dajoRH commented:
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

dajoRH commented:
I detected some fixture changes in commit ce670521f5e8f6c72ce01975f57429860ae1915a

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`create_vm_modscope`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**
    - *policy_for_testing*
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *alert_profile*
    - *test_embedded_ansible_custom_button_localhost*
    - *test_embedded_ansible_custom_button_target_machine*
    - *test_embedded_ansible_custom_button_specific_hosts*
    - *test_alert_run_ansible_playbook*

The *global* fixture **`create_vm`** is used in the following files:
- **cfme/tests/automate/test_service_automate.py**
    - *test_retire_vm_now*
- **cfme/tests/control/test_alerts.py**
    - *wait_candu*
    - *ssh*
    - *test_alert_rtp*
    - *test_alert_timeline_cpu*
    - *test_alert_snmp*
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *test_service_dialog_expression_method*
- **cfme/tests/webui/test_general_ui.py**
    - *test_vm_right_size_recommendation_back_button*

The *local* fixture **`policy_for_testing`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_actions.py**

The *local* fixture **`alert_profile`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_automate.py**
    - *test_alert_run_ansible_playbook*

The *local* fixture **`wait_candu`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

The *local* fixture **`ssh`** is used in the following files:
- **cfme/tests/control/test_alerts.py**

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:
